railib/rest.py

```diff
@@ -119,13 +121,52 @@ def _print_request(req: Request, level=0):
                 print(json.dumps(json.loads(req.data.decode("utf8")), indent=2))
 
 
+def _cache_file() -> str:
+    return path.join(path.expanduser('~'), '.rai', 'tokens.json')
+
+
+# Read oauth cache
+def _read_cache() -> dict:
+    try:
+        with open(_cache_file(), 'r') as cache:
+            return json.loads(cache.read())
+    except Exception:
+        return None
+
+
+# Read access token from cache
+def _read_token_cache(creds: ClientCredentials) -> AccessToken:
+    try:
+        cache = _read_cache()
+        return AccessToken(**cache[creds.client_id])
+    except Exception:
+        return None
+
+
+# write access token to cache
+def _write_token_cache(creds: ClientCredentials):
+    try:
+        cache = _read_cache()
+        if cache:
+            cache[creds.client_id] = creds.access_token
+        else:
+            cache = {creds.client_id: creds.access_token}
```

Comment:
maybe we could avoid this branch by returning `{}` from `_read_cache`'s exception path